File: client/src/app/models/product.ts
export interface Product {
    id: number;
    name: string;
    description: string;
    price: number;
    pictureUrl: string;
    brand?: string;
    color?: string;
    category?: string;
    quantityInStock?: number;
    createdDate: Date;
    updatedDate: Date;
}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    colors: string[];
    brands: string[];
    categories: string[];
    pageNumber: number;
    pageSize: number;
}

File: api/src/ReStore.Domain/Entities/Product.cs
namespace ReStore.Domain.Entities;

public class Product : BaseEntity
{
     public string Name { get; set; }
     public string Description { get; set; }
     public long Price { get; set; }
     public string PictureUrl { get; set; }
     public string Category { get; set; }
     public string Color { get; set; }
     public string Brand { get; set; }
     public int QuantityInStock { get; set; }

     public string? PublicId { get; set; }
}
